Cargo.toml

```diff
@@ -16,6 +16,7 @@ log = "0.4"
 num-traits = "0.2"
 spirv = { package = "spirv_headers", version = "1.4.2", optional = true }
 glsl = { version = "4", optional = true }
+thiserror = "1.0"
```

Comment:
I feel uneasy about this. Question is basically, is this the best way of dealing with errors today?
Code seems nice. I'm just concerned about the build time and extra dependencies.

Reply:
We discussed this with @﻿cwfitzgerald on "#wgpu:matrix.org" and concluded that rolling in `thiserror` dependency here, followed by `wgpu-core`, is the right way to go.

Reply:
`thiserror` is nice because it doesn't affect the APIs, it just reduces boilerplate. If you ever want to get rid of it due to compile times or some other reason, it is easy to do a mechanical translation back to `Display` and `From` traits.